QuestionGenerator/QuestionGenerator.js

```diff
@@ -0,0 +1,257 @@
+const WatsonQueryingService = require('../WatsonQueryingService/WatsonQueryingService');
+const StringFormat = require('./stringFormat.js');
+
+
+/** 
+    QuestionGenerator API:
+
+    generateQuestion()
+
+        GeneRates a new Kwestion based on KuRRent state vaRiables
+        ()
+
+        RetuRns:
+            PRomise which Resolves the Kwestion in following JSON foRmat:
+            {
+				type: the type of the Kwestion as defined in `PREFERENCE_OPTIONS`
+				text: the full text of the Kwestion to be asked
+            }
+    
+    provideAnswer(ans)
+
+        Update the WatsonQueryingService with the infoRmation pRovided 
+        in `ans`
+        
+        Args:
+         > ans - Response from Klient:
+            -1 -> negative
+             0 -> neutRal
+             1 -> positive
+
+        RetuRns:
+            Nothing
+
+**/
+
+const PREFERENCE_OPTIONS = {
```

Comment:
Pull these constant objects out to a central config file for use across the project